### ISSUE TITLE
Terminal apps can't be opened with certain terminals

### ISSUE THREAD

**Issue by astier** [linuxmint/cinnamon#8246]
```
 * Cinnamon version 4.0.8
 * Distribution - Arch
 * Graphics hardware *and* driver: Intel HD Graphics 5300, Modesetting
 * 64 bit
 ```

**Issue**

Terminal-applications like htop and neovim can not be opened with the termite- and urxvt-terminals when one of this terminals is set as the default terminal in the *preferred applications menu*. Opening from within the terminal is possible (open terminal and then type htop or neovim as command). However, neovim and htop do not open when one clicks at the respective icons in the main-menu. Also text-files which should be opened with neovim do not open. This is possible with other terminals like xterm or gnome-terminal.

**Steps to reproduce**

1. Install termite/rxvt-unicode
2. Install htop/neovim
2. Uninstall gnome-terminal and xterm
3. Set termite/urxvt as the default terminal in the preferred-applications-menu
4. Set neovim as the default application for plain text files
4. Click on the htop/neovim-icon in the main-menu
5. Click on a plain-text-file
6. Observe that nothing happens and the application/text-file does not open

**Expected behavior**

The respective terminal-application and text-file should open using the default terminal (termite/urxvt).

**Comment by astier** [comment]
Please let me know if there is a workaround.

**Comment by icarter09** [comment]
@astier thanks for referencing the other ticket. I'm going to start looking into this issue and putting together a PR.

**Comment by astier** [comment]
@icarter09 is there a chance that this will be fixed in the near future? Right now people are kind of forced to use the gnome-terminal.

**Comment by astier** [comment]
I have tried it neither on Linux Mint nor in Cinnamon but hard-linking the desired terminal to xterm solves the issue for me. I think this is a deeper issue which is not caused by Cinnamon. After searching on the Internet I found people from other dustributions with similar issues. The hard-link approach was the only one which worked for me. For the st terminal it would be:

```sh
sudo ln -f /usr/local/bin/st /usr/bin/xterm
```
